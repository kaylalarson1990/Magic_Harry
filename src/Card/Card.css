.card {
    align-items: center;
    background-color: white;
    border: .5px solid #EBECEC;
    border-radius: 3px;
    box-shadow: 2px 2px lightgrey;
    display: flex;
    flex-direction: column;
    font-family: 'Open Sans';
    height: auto;
    margin: 15px;
    width: 300px;
}

button,
.favorite-btn {
    border: 1px solid black;
    border-radius: 20px;
    cursor: pointer;
    font-size: 12px;
    margin: 25px 0;
    padding: 10px;
    width: 180px;
}

button:hover {
    background-color: #ECBA16
}

span {
    font-weight: 700;
}


h2 {
    text-align: center;
    font-weight: 600;
}

.favorite {
    background-color: #ECBA16;
}

