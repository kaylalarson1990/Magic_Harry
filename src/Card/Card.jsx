import React, { Component } from "react";
import "./Card.css";
import ReactCardFlip from "react-card-flip";
import PropTypes from "prop-types";

export class Card extends Component {
  constructor() {
    super();
    this.state = {
      isFlipped: false
    };
  }

  handleClick = e => {
    e.preventDefault();
    this.setState(prevState => ({ isFlipped: !prevState.isFlipped }));
  };

  render() {
    const { favoriteCard, info } = this.props;
    const cardInfo = [];
    for (let key in info) {
      if (key !== "id" && key !== "favorite") {
        cardInfo.push(
          <p className="card-info">
            <span className="head">{key.toUpperCase()}:</span> {info[key]}
          </p>
        );
      }
    }

    return (
      <ReactCardFlip isFlipped={this.state.isFlipped} flipDirection="vertical">
        <article className="card" key="front">
          <h2>{info.name || info.spell}</h2>
          <button className="flip-card1" onClick={this.handleClick}>
            Click here for more details
          </button>
          <button
            onClick={e => favoriteCard(info.id)}
            className={`favorite-btn ${info.favorite && "favorite"}`}
          >
            Favorite
          </button>
        </article>
        <article className="card" key="back">
          {cardInfo}
          <button className="flip-card2" onClick={this.handleClick}>
            Return to Front
          </button>
        </article>
      </ReactCardFlip>
    );
  }
}

Card.propTypes = {
  info: PropTypes.object,
  favoriteCard: PropTypes.func
};

export default Card;
