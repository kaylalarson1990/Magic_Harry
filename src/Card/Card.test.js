import React from "react";
import { shallow } from "enzyme";
import Card from "./Card.jsx";

describe("Card", () => {
  let mockFunc, mockData, instance, wrapper;
  let event = {
    preventDefault: jest.fn()
  }

  beforeEach(() => {
    mockFunc = jest.fn();
    mockData = {
      name: "Kayla",
      id: "1",
      role: "Student",
      house: "Apartment",
      school: "Turing",
      species: "Human"
    };
    wrapper = shallow(<Card info={mockData} favoriteCard={mockFunc} />);

    instance = wrapper.instance()
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should set isFlipped state", () => {
    const wrapper = shallow(<Card info={mockData} favoriteCard={mockFunc} />);

    expect(wrapper.state('isFlipped')).toEqual(false)
    instance.handleClick(event);
    expect(wrapper.state('isFlipped')).toEqual(false)

  });

  it("should call favorite card on click", () => {
    const wrapper = shallow(<Card info={mockData} favoriteCard={mockFunc} />);

    mockFunc("1");
    wrapper.find(".favorite-btn").simulate("click");
    expect(mockFunc).toBeCalledWith(mockData.id);
  });
});
