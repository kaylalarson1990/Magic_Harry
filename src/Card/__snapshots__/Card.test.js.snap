// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<t
  cardStyles={
    Object {
      "back": Object {},
      "front": Object {},
    }
  }
  cardZIndex="auto"
  containerStyle={Object {}}
  flipDirection="vertical"
  flipSpeedBackToFront={0.6}
  flipSpeedFrontToBack={0.6}
  infinite={false}
  isFlipped={false}
>
  <article
    className="card"
    key="front"
  >
    <h2>
      Kayla
    </h2>
    <button
      className="flip-card1"
      onClick={[Function]}
    >
      Click here for more details
    </button>
    <button
      className="favorite-btn undefined"
      onClick={[Function]}
    >
      Favorite
    </button>
  </article>
  <article
    className="card"
    key="back"
  >
    <p
      className="card-info"
    >
      <span
        className="head"
      >
        NAME
        :
      </span>
       
      Kayla
    </p>
    <p
      className="card-info"
    >
      <span
        className="head"
      >
        ROLE
        :
      </span>
       
      Student
    </p>
    <p
      className="card-info"
    >
      <span
        className="head"
      >
        HOUSE
        :
      </span>
       
      Apartment
    </p>
    <p
      className="card-info"
    >
      <span
        className="head"
      >
        SCHOOL
        :
      </span>
       
      Turing
    </p>
    <p
      className="card-info"
    >
      <span
        className="head"
      >
        SPECIES
        :
      </span>
       
      Human
    </p>
    <button
      className="flip-card2"
      onClick={[Function]}
    >
      Return to Front
    </button>
  </article>
</t>
`;
